version: '3.8'
services:
  pwf-db:
    image: postgres:14.1-alpine
    container_name: pwf-db
    restart: always
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
    ports:
      - '54320:54320'
    volumes:
      - pwf-db:/var/lib/postgresql/data
  pwf-db-pgadmin:
    image: dpage/pgadmin4
    logging:
      driver: none
    container_name: pwf-db-pgadmin
    restart: always
    ports:
      - '5050:80'
    environment:
      PGADMIN_DEFAULT_EMAIL: admin@admin.com
      PGADMIN_DEFAULT_PASSWORD: admin
    volumes:
      - pgadmin-data:/var/lib/pgadmin
  pwf-api:
    container_name: 'pwf-api'
    build: ./api
    volumes:
      - ./api/.build:/pwf-api/.build
      - ./api/src:/pwf-api/src
      - ./api/test:/pwf-api/test
      - ./api/.env:/pwf-api/.env
    command: npm run start:dev
    ports:
      - '3000:3002'
    extra_hosts:
      - 'host.docker.internal:host-gateway'
  pwf-seed:
    container_name: 'pwf-seed'
    depends_on:
      - pwf-db
      - pwf-api
    build: ./seed/.
    command: python3 /seed/seed.py
    volumes:
      - type: bind
        source: ./seed/seeding
        target: /seed
volumes:
  pwf-db:
  pgadmin-data:
  pwf-api:
networks:
  default:
    name: sl-manager-local_default
